uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/leave-request-popup-controller.js

```diff
@@ -563,12 +608,20 @@ define([
        * Sets dates and types for vm.leaveRequest from UI
        */
       function setDateAndTypes() {
-        if (!vm.uiOptions.multipleDays) {
-          vm.leaveRequest.to_date_type = vm.leaveRequest.from_date_type;
-        }
+        vm.leaveRequest.from_date = vm.uiOptions.fromDate ? convertDateFormatToServer(vm.uiOptions.fromDate) : null;
+        vm.leaveRequest.to_date = vm.uiOptions.toDate ? convertDateFormatToServer(vm.uiOptions.toDate) : null;
+
+        if (vm.uiOptions.multipleDays) {
+          vm.uiOptions.showBalance = !!vm.leaveRequest.to_date && !!vm.leaveRequest.from_date;
+        } else {
```

Comment:
`} else if (vm.uiOptions.fromDate) {` to increase readability and avoid Pyramid of doom